modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Jetpack Search: Instant Front-End Search and Filtering
+ *
+ * @since 8.3.0
+ * @package jetpack
+ */
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Constants;
+
+class Jetpack_Instant_Search extends Jetpack_Search {
+
+	/**
+	 * Jetpack_Instant_Search constructor.
+	 *
+	 * @since 5.0.0
+	 *
+	 * Doesn't do anything. This class needs to be initialized via the instance() method instead.
+	 */
+	protected function __construct() {
+	}
+
+	/**
+	 * Loads the php for this version of search
+	 *
+	 * @since 8.3.0
+	 */
+	public function load_php() {
+		require_once dirname( __FILE__ ) . '/class.jetpack-search-template-tags.php';
+		require_once JETPACK__PLUGIN_DIR . 'modules/widgets/search.php';
+		require_once dirname( __FILE__ ) . '/class.jetpack-search-template-tags.php';
+
+		if ( class_exists( 'WP_Customize_Manager' ) ) {
+			require_once dirname( __FILE__ ) . '/class-jetpack-search-customize.php';
+			new Jetpack_Search_Customize();
+		}
+	}
+
+	/**
+	 * Setup the various hooks needed for the plugin to take over search duties.
+	 *
+	 * @since 5.0.0
+	 */
+	public function init_hooks() {
+		if ( ! is_admin() ) {
+			add_filter( 'posts_pre_query', array( $this, 'filter__posts_pre_query' ), 10, 2 );
+			add_action( 'parse_query', array( $this, 'action__parse_query' ), 10, 1 );
+
+			add_action( 'init', array( $this, 'set_filters_from_widgets' ) );
+
+			add_action( 'wp_enqueue_scripts', array( $this, 'load_assets' ) );
+		} else {
+			add_action( 'update_option', array( $this, 'track_widget_updates' ), 10, 3 );
+		}
+
+		add_action( 'jetpack_deactivate_module_search', array( $this, 'move_search_widgets_to_inactive' ) );
+	}
+
+	/**
+	 * Loads assets for Jetpack Instant Search Prototype featuring Search As You Type experience.
+	 */
+	public function load_assets() {
+		$script_relative_path = '_inc/build/instant-search/jp-search.bundle.js';
+		if ( ! file_exists( JETPACK__PLUGIN_DIR . $script_relative_path ) ) {
```

Comment:
Should we do checks for both the script and the style here? The overlay won't work without the stylesheets.